CIP-0072/README.md

```diff
@@ -0,0 +1,413 @@
+
+## **Specification**
+
+### **Developers / Publishers**
+Developers and publishers of dApps can register their dApps by submitting a transaction on-chain that can be indexed and verifier by stores, auditors and other actors. 
+
+### **Stores / Auditors**
+Store and auditors should be able to follow the chain and find when a new dApp registration is **anchored** on-chain. They should then perform *integrity* and *trust* validations on the dApp's certificate and metadata. 
+
+#### **Off-chain Location Adversitment**
+Each store and auditor should make public the location of their off-chain sources where they will look for the dApp's metadata based on certificates found on-chain. These can be advertised through their own API or publically available on a website.
+
+Sample off-chain sources could be for example :
+- [CIP-26](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0026) compliant servers
+- IPFS
+
+##### **Targetted Releases**
+Each developer and publisher can choose to where to write metadata based on the information available from known stores & auditors. This gives **developers** and **publishers** the ability to perform targeted releases. (i.e to which stores and auditors)
+
+#### **Suggested Validations**
+
+- **`integrity`**: The dapp's metadata off-chain should match the metadata anchored on-chain.
+- **`trust`**: The dApp's certificate should be signed by a trusted entity. It's up to the store/auditor to decide which entities are trusted and they should maintain their own list of trusted entities. Although this entities might be well known, it's not the responsibility of this CIP to maintain this list.
+
+### **On-chain dApp Registration certificate**
+
+```json
+{
+	"subject": "d684512ccb313191dd08563fd8d737312f7f104a70d9c72018f6b0621ea738c5b8213c8365b980f2d8c48d5fbb2ec3ce642725a20351dbff9861ce9695ac5db8",
+	"rootHash": "8c4e9eec512f5f277ab811ba75c991d51600c80003e892e601c6b6c19aaf8a33",
+  "schema_version": "0.0.1",
+  "type": { 
+    "action": "REGISTER",
+    "releaseNumber": "1.0.0",
+    "releaseName": "My First Release",
+  },
+	"signature": {
+		"r": "5114674f1ce8a2615f2b15138944e5c58511804d72a96260ce8c587e7220daa90b9e65b450ff49563744d7633b43a78b8dc6ec3e3397b50080",
+		"s": "a15f06ce8005ad817a1681a4e96ee6b4831679ef448d7c283b188ed64d399d6bac420fadf33964b2f2e0f2d1abd401e8eb09ab29e3ff280600",
+		"algo": "Ed25519−EdDSA",
+		"pub": "b7a3c12dc0c8c748ab07525b701122b88bd78f600c76342d27f25e5f92444cde"
+	}
+}
+```
+
+### Properties
+*`subject`*: Identifier of the claim subject (i.e dapp). A UTF-8 encoded string. This uniquess of this property cannot be guaranteed by the protocol and multiple claims for the same subject may exist therefore it is required to exist some mechanism to assert trust in the *veracity* of this property.
+
+*`type`*: The type of the claim. This is a JSON object that contains the following properties: 
+- *`action`*: The action that the certificate is asserting. It can take the following values: 
```

Comment:
Will we also add `CERTIFICATION` here? (Though there's a horrible ambiguity in usage here; not your fault.)